### PR TITLE
add tests for MessageCryptoHelper

### PR DESCRIPTION
**Pull request by Valodim**
This class has been criminally undertested, which came back to bite me in the autocrypt branches once too often. So here's a bunch of tests to check basic functionality. Also includes a bunch of related changes to make testing easier.

### CODE REVIEW

File: k9mail-library/src/main/java/com/fsck/k9/mail/Message.java [hunk 1]
```diff
@@ -57,7 +57,7 @@ public int hashCode() {
         final int MULTIPLIER = 31;
 
         int result = 1;
-        result = MULTIPLIER * result + mFolder.getName().hashCode();
+        result = MULTIPLIER * result + (mFolder != null ? mFolder.getName().hashCode() : 0);
```
Comment (cketti) [inline k9mail-library/src/main/java/com/fsck/k9/mail/Message.java]: Can `mFolder == null` happen during regular operations? Or is it only so we can be lazy in tests?

File: k9mail/src/main/java/com/fsck/k9/crypto/MessageDecryptVerifier.java [hunk 2]
```diff
@@ -200,11 +200,9 @@ private static Part findPrimaryPartInAlternative(Part part) {
             if (body instanceof Multipart) {
                 Multipart multi = (Multipart) body;
                 BodyPart signatureBody = multi.getBodyPart(1);
-                if (isSameMimeType(signatureBody.getMimeType(), APPLICATION_PGP_SIGNATURE)) {
-                    ByteArrayOutputStream bos = new ByteArrayOutputStream();
-                    signatureBody.getBody().writeTo(bos);
-                    return bos.toByteArray();
-                }
+                ByteArrayOutputStream bos = new ByteArrayOutputStream();
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/crypto/MessageDecryptVerifier.java]: Why did you remove the MIME type check?

Please create a separate commit for such changes. That way you can write down what motivated this.

File: k9mail-library/src/main/java/com/fsck/k9/mail/Message.java (hunk 1 above)
Comment (Valodim) [reply k9mail-library/src/main/java/com/fsck/k9/mail/Message.java]: Little bit of both. A typical newly constructed MimeMessage instance won't have a folder yet, and iirc there isn't even a way to set one in MimeMessage itself.

File: k9mail/src/main/java/com/fsck/k9/crypto/MessageDecryptVerifier.java (hunk 2 above)
Comment (Valodim) [reply k9mail/src/main/java/com/fsck/k9/crypto/MessageDecryptVerifier.java]: Sorry, separate commit would have been the way to go. This check should happen before a part is even considered for parsing as pgp/mime signed, if it doesn't match here it means we screwed up before
